src/style.rs

```diff
@@ -315,211 +474,205 @@ impl<T> From<T> for StyleValue<T> {
     }
 }
 
-// Creates `ComputedStyle` which has definite values for the fields, barring some specific cases.
-// Creates `Style` which has `StyleValue<T>`s for the fields
-macro_rules! define_styles {
+/// A style with definite values for most fields.
+#[derive(Default, Debug, Clone)]
+pub struct ComputedStyle {
+    pub(crate) other: StyleMap,
+}
+impl ComputedStyle {
+    pub(crate) fn get<P: StyleProp>(&self, _prop: P) -> P::Type {
+        self.other
+            .get_prop::<P>()
+            .unwrap_or_else(|| P::default_value())
+    }
+
+    pub(crate) fn get_builtin(&self) -> BuiltinStyleReader<'_> {
+        BuiltinStyleReader { style: self }
+    }
+}
+#[derive(Debug, Clone)]
+pub struct Style {
```

Comment:
I'm wondering if it's better to move StyleMap content here. e.g.

#[derive(Default, Clone)]
pub struct Style {
    pub(crate) map: HashMap<StylePropRef, StyleMapValue<Rc<dyn Any>>>,
    pub(crate) selectors: HashMap<StyleSelector, StyleMap>,
}

Reply:
There's quite a bit of stuff that can be refactored with this PR. I feel like the PR is sufficiently large already though.

Reply:
Yeah we can merge this first and work on stuff in another PR.